bridge-impl/src/main/java/com/liferay/faces/bridge/scope/internal/BridgeRequestScopeManagerImpl.java

```diff
@@ -39,105 +43,164 @@
 	// Logger
 	private static final Logger logger = LoggerFactory.getLogger(BridgeRequestScopeManagerImpl.class);
 
-	public void removeBridgeRequestScopesByPortlet(PortletConfig portletConfig) {
-		String portletNameToRemove = portletConfig.getPortletName();
-		PortletContext portletContext = portletConfig.getPortletContext();
-		BridgeRequestScopeCache bridgeRequestScopeCache = BridgeRequestScopeCacheFactory
-			.getBridgeRequestScopeCacheInstance(portletContext);
-		Set<Map.Entry<String, BridgeRequestScope>> mapEntries = bridgeRequestScopeCache.entrySet();
+	// Private Constants
+	private static final String ATTR_BRIDGE_REQUEST_SCOPE_CACHE = "com.liferay.faces.bridge.bridgeRequestScopeCache";
+
+	@Override
+	public Cache<String, BridgeRequestScope> getBridgeRequestScopeCache(PortletContext portletContext) {
+
+		Cache<String, BridgeRequestScope> bridgeRequestScopeCache;
+
+		synchronized (portletContext) {
```

Comment:
@stiemannkj1 There is no guarantee that the META-INF/faces-config.xml files will have been scanned by the time `Bridge.init(PortletConfig)` is called. In practice, Liferay Portal and Pluto behave differently in this regard. I can't remember which one it is, but one of them finishes ServletContext initialization (and hence finishes scanning for META-INF/faces-config.xml files) before `Bridge.init(PortletConfig)` is called and the other doesn't. For that reason, a lot of creation of factory instances is deferred until the bridge implementation has received the first request using the thread-safe double-check idom. For additional info, CTRL+F for the word "defer" in [FACES-3106](https://issues.liferay.com/browse/FACES-3106) and [FACES-2778](https://issues.liferay.com/browse/FACES-2778).